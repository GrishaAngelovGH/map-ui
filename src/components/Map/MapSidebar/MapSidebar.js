import { Fragment } from 'react'
import Sidebar from '../../Sidebar'
import Place from './Place'

const MapSidebar = ({ showSidebar, onHide }) => {
  const places = [
    { title: "Bill Baker's Street Restaurant", open: true },
    { title: 'Pizza Restaurant Waterloo', open: false },
    { title: 'Leicester Square Restaurant', open: true },
    { title: 'Mayfair Restaurant', open: true },
    { title: 'Chelsea Restaurant', open: true }
  ]

  return (
    <Sidebar
      title='Favorite Places'
      body={
        <Fragment>
          {
            places.map((v, i) => (
              <Place key={i} title={v.title} open={v.open} />
            ))
          }
        </Fragment>
      }
      showSidebar={showSidebar}
      onHide={onHide}
    />
  )
}

export default MapSidebar