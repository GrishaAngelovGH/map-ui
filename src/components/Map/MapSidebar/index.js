import MapSidebar from './MapSidebar'

export default MapSidebar