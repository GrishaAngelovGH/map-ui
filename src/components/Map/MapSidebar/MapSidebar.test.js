import { render } from '@testing-library/react'

import MapSidebar from './MapSidebar'

window.matchMedia = window.matchMedia || function () {
  return {
    matches: false,
    addListener: function () { },
    removeListener: function () { }
  }
}

test('should render MapSidebar component', () => {
  const view = render(<MapSidebar showSidebar />)

  expect(view).toMatchSnapshot()
})