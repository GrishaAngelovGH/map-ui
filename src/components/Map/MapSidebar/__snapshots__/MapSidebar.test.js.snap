// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MapSidebar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      aria-modal="true"
      class="offcanvas offcanvas-start offcanvas-toggling show"
      role="dialog"
      style="visibility: visible;"
      tabindex="-1"
    >
      <div
        class="offcanvas-header"
      >
        <div
          class="offcanvas-title h5"
        >
          Favorite Places
        </div>
        <button
          aria-label="Close"
          class="btn-close"
          type="button"
        />
      </div>
      <div
        class="offcanvas-body"
      >
        <div
          class="row mb-3 border border-3 rounded shadow p-3 bg-light small"
        >
          <div
            class="col-md-10"
          >
            <p
              class="fw-bold text-center"
            >
              Bill Baker's Street Restaurant
            </p>
            <p
              class="p-0 alert alert-success text-center"
            >
              Open
            </p>
            <p
              class="m-0 text-secondary"
            >
              4.3 
              <i
                class="bi bi-star-fill text-warning"
              />
               (3037 votes)
            </p>
            <p
              class="m-0 text-secondary"
            >
              Contemporary all-day European dining
            </p>
            <p
              class="m-0 text-secondary"
            >
              Closes
               11 PM
            </p>
          </div>
          <div
            class="col-md-2"
          >
            <i
              class="bi bi-check-square-fill text-primary fs-1"
            />
          </div>
        </div>
        <div
          class="row mb-3 border border-3 rounded shadow p-3 bg-light small"
        >
          <div
            class="col-md-10"
          >
            <p
              class="fw-bold text-center"
            >
              Pizza Restaurant Waterloo
            </p>
            <p
              class="p-0 alert alert-danger text-center"
            >
              Closed
            </p>
            <p
              class="m-0 text-secondary"
            >
              4.3 
              <i
                class="bi bi-star-fill text-warning"
              />
               (3037 votes)
            </p>
            <p
              class="m-0 text-secondary"
            >
              Contemporary all-day European dining
            </p>
            <p
              class="m-0 text-secondary"
            >
              Opens
               11 PM
            </p>
          </div>
          <div
            class="col-md-2"
          >
            <i
              class="bi bi-check-square-fill text-primary fs-1"
            />
          </div>
        </div>
        <div
          class="row mb-3 border border-3 rounded shadow p-3 bg-light small"
        >
          <div
            class="col-md-10"
          >
            <p
              class="fw-bold text-center"
            >
              Leicester Square Restaurant
            </p>
            <p
              class="p-0 alert alert-success text-center"
            >
              Open
            </p>
            <p
              class="m-0 text-secondary"
            >
              4.3 
              <i
                class="bi bi-star-fill text-warning"
              />
               (3037 votes)
            </p>
            <p
              class="m-0 text-secondary"
            >
              Contemporary all-day European dining
            </p>
            <p
              class="m-0 text-secondary"
            >
              Closes
               11 PM
            </p>
          </div>
          <div
            class="col-md-2"
          >
            <i
              class="bi bi-check-square-fill text-primary fs-1"
            />
          </div>
        </div>
        <div
          class="row mb-3 border border-3 rounded shadow p-3 bg-light small"
        >
          <div
            class="col-md-10"
          >
            <p
              class="fw-bold text-center"
            >
              Mayfair Restaurant
            </p>
            <p
              class="p-0 alert alert-success text-center"
            >
              Open
            </p>
            <p
              class="m-0 text-secondary"
            >
              4.3 
              <i
                class="bi bi-star-fill text-warning"
              />
               (3037 votes)
            </p>
            <p
              class="m-0 text-secondary"
            >
              Contemporary all-day European dining
            </p>
            <p
              class="m-0 text-secondary"
            >
              Closes
               11 PM
            </p>
          </div>
          <div
            class="col-md-2"
          >
            <i
              class="bi bi-check-square-fill text-primary fs-1"
            />
          </div>
        </div>
        <div
          class="row mb-3 border border-3 rounded shadow p-3 bg-light small"
        >
          <div
            class="col-md-10"
          >
            <p
              class="fw-bold text-center"
            >
              Chelsea Restaurant
            </p>
            <p
              class="p-0 alert alert-success text-center"
            >
              Open
            </p>
            <p
              class="m-0 text-secondary"
            >
              4.3 
              <i
                class="bi bi-star-fill text-warning"
              />
               (3037 votes)
            </p>
            <p
              class="m-0 text-secondary"
            >
              Contemporary all-day European dining
            </p>
            <p
              class="m-0 text-secondary"
            >
              Closes
               11 PM
            </p>
          </div>
          <div
            class="col-md-2"
          >
            <i
              class="bi bi-check-square-fill text-primary fs-1"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
